src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -829,6 +830,103 @@ ov::pass::NopSliceBeforeGatherElements::NopSliceBeforeGatherElements() {
     register_matcher(m, matcher_pass_callback);
 }
 
+ov::pass::NopStridedSlice::NopStridedSlice() {
+    MATCHER_SCOPE(NopStridedSlice);
+
+    auto input = pattern::any_input();
+    auto begin_const = pattern::wrap_type<ov::op::v0::Constant>();
+    auto end_const = pattern::wrap_type<ov::op::v0::Constant>();
+    auto optional_stride_const = pattern::optional<ov::op::v0::Constant>();
+    auto pattern = pattern::wrap_type<ov::op::v1::StridedSlice>({input, begin_const, end_const, optional_stride_const});
+
+    ov::matcher_pass_callback matcher_pass_callback = [=](pattern::Matcher& m) {
+        auto node = m.get_match_root();
+        if (node == nullptr) {
+            return false;
+        }
+        auto strided_slice_node = std::dynamic_pointer_cast<ov::op::v1::StridedSlice>(node);
+        if (!strided_slice_node->get_shrink_axis_mask().empty() || !strided_slice_node->get_new_axis_mask().empty() ||
+            !strided_slice_node->get_ellipsis_mask().empty()) {
+            return false;
+        }
+        if (!op::util::is_constant_and_all_values_equal_int(node->input_value(3), 1) &&
```

Comment:
why do we need to check node->get_input_partial_shape(3).size() != 0?

please check that 4rd input exists before using it here "node->input_value(3)":
if (node->get_input_size() == 4 && !op::util::is_constant_and_all_values_equal_int(node->input_value(3), 1))

Reply:
fixed